resources/ssh.rb

```diff
@@ -0,0 +1,38 @@
+#
+# Cookbook Name:: artifactory_pro
+# Resource:: ssh
+#
+# Copyright (C) 2015 Monkey Little
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+
+actions [:add, :update, :remove]
+
+default_action :add
+
+attribute :host, kind_of: String
+attribute :hostname, kind_of: String
+attribute :port, kind_of: Integer
+attribute :key, kind_of: String
+attribute :type, equal_to: %w(dsa rsa ecdsa), default: 'rsa'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.